lift/src/main/scala/com/linkedin/lift/lib/PositionBiasUtils.scala

```diff
@@ -0,0 +1,191 @@
+package com.linkedin.lift.lib
+
+import com.linkedin.lift.types.ScoreWithLabelAndPosition
+import org.apache.spark.mllib.stat.KernelDensity
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.expressions.Window
+import org.apache.spark.sql.functions.{broadcast, lit, min, rand}
+import org.apache.spark.sql.{Dataset, SparkSession}
+
+object PositionBiasUtils {
+
+  case class PositionBias(position: Int, positionBias: Double)
+
+  /**
+    * Bandwidth computation based on Silverman's rule for kernel density estimation
+    *
+    * @param data i.i.d. samples
+    * @return bandwidth
+    */
+  def getBandwidth(data: RDD[Double]): Double = {
+    val numSamples = data.count
+    val mean = data.sum / numSamples
+    val stdDev = Math.sqrt((data.map(t => t * t).sum / numSamples) - mean * mean)
+    return 1.06 * stdDev * math.pow(numSamples, -0.2)
+  }
+
+  /**
+    * Kernel density estimation with a Gaussian kernel
+    *
+    * @param data i.i.d. samples
+    * @return a probability density function
+    */
+  def getDensity(data: RDD[Double]): KernelDensity = {
+    val bw = getBandwidth(data)
+    val density = new KernelDensity()
+      .setSample(data)
+      .setBandwidth(bw)
+    return density
+  }
+
+  /**
+    * Extracting score corresponding to a given position from a Dataset
+    *
+    * @param spark    spark session
+    * @param data     dataset containing score, binary label/response, position
+    * @param position position value for filtering
+    *                 return score RDD
+    */
+  def extractScore(spark: SparkSession, data: Dataset[ScoreWithLabelAndPosition], position: Int): RDD[Double] = {
+    import spark.implicits._
+    return data
+      .filter($"position" === position)
```

Comment:
Since you're working with the Spark Dataset API, to keep this filter type-safe, you can do something like this instead: `.filter(_.position == position)`